### PR TITLE
Feat/#6 create the repo interface and the service class

### LINKED ISSUE #6: Create the Repo interface and  the Service class
The **Repo** interface will be used in order to send queries to the **DB** through **JPA**.\
The **Service** class will contain the *logic* of the application.\
An example can be found in the [demo project](https://github.com/scoala-online/demo-project)

### PR DESCRIPTION
**Pull request by sebastianbadea197**
Add the repo and service classes.

Closes #6

### CODE REVIEW

File: apps/java-api/src/main/java/org/scoalaonline/api/model/Department.java [hunk 1]
```diff
@@ -38,7 +38,7 @@ public long getId() {
    * Getter for the name of the department
    * @return the name of the department
    */
-  public String getDepartmentName() {
+  public String getName() {
```
Comment (CristiPV) [inline apps/java-api/src/main/java/org/scoalaonline/api/model/Department.java]:
```suggestion
  public String getDepartmentName() {
```

File: apps/java-api/src/main/java/org/scoalaonline/api/model/Department.java [hunk 2]
```diff
@@ -60,7 +60,7 @@ public void setId(long id) {
   /**
    * Setter for the name of the department
    */
-  public void setDepartmentName(String departmentName) {
+  public void setName(String departmentName) {
```
Comment (CristiPV) [inline apps/java-api/src/main/java/org/scoalaonline/api/model/Department.java]:
```suggestion
  public void setDepartmentName(String departmentName) {
```

File: apps/java-api/src/main/java/org/scoalaonline/api/service/CorporatistService.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+   */
+  public Corporatist addCorporatist (Corporatist corporatist) {
+    Corporatist corporatistToAdd = new Corporatist();
+    corporatistToAdd.setFirstName(corporatist.getFirstName());
+    corporatistToAdd.setMiddleName(corporatist.getMiddleName());
+    corporatistToAdd.setLastName(corporatist.getLastName());
+    corporatistToAdd.setEmail(corporatist.getEmail());
+    corporatistToAdd.setDepartment(corporatist.getDepartment());
+    corporatistToAdd.setSupervisor(corporatist.getSupervisor());
+    corporatistToAdd.setId(corporatist.getId());
```
Comment (CristiPV) [inline apps/java-api/src/main/java/org/scoalaonline/api/service/CorporatistService.java]:
```suggestion
```

File: apps/java-api/src/main/java/org/scoalaonline/api/service/CorporatistService.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+   * @param corporatist which is the new entity to be put into the repo.
+   * @return the repo with the changed entity.
+   */
+  public Corporatist updateCorporatist (long id, Corporatist corporatist) {
+    try {
+      Corporatist corporatistToUpdate = corporatistRepo.findById(id).get();
+      if (corporatist.getFirstName() != null) {
+        corporatistToUpdate.setFirstName(corporatist.getFirstName());
+      }
+        corporatistToUpdate.setMiddleName(corporatist.getMiddleName());
```
Comment (CristiPV) [inline apps/java-api/src/main/java/org/scoalaonline/api/service/CorporatistService.java:69]:
```suggestion
      if (corporatist.getMiddleName() != null) {
        corporatistToUpdate.setMiddleName(corporatist.getMiddleName());
      }
```

**Comment by sebastianbadea197** [comment]
🚀🚀🚀

File: apps/java-api/src/main/java/org/scoalaonline/api/service/DepartmentService.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+  /**
+   * Adds a department entity
+   * @param department the department to be added
+   * @return the new configuration with the department added
+   */
+  public Department addDepartment ( Department department ) {
+    Department departmentToAdd = new Department();
+    departmentToAdd.setName( department.getName() );
+    departmentToAdd.setId( department.getId() );
```
Comment (CristiPV) [inline apps/java-api/src/main/java/org/scoalaonline/api/service/DepartmentService.java]:
```suggestion
```

File: apps/java-api/src/main/java/org/scoalaonline/api/service/SupervisorService.java [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+  /**
+   * Adds a new supervisor into the database.
+   * @param supervisor which is the supervisor to be added.
+   * @return the repo with the supervisor added.
+   */
+  public Supervisor addSupervisor (Supervisor supervisor) {
+    Supervisor supervisorToAdd = new Supervisor();
+
+    supervisorToAdd.setId(supervisor.getId());
```
Comment (CristiPV) [inline apps/java-api/src/main/java/org/scoalaonline/api/service/SupervisorService.java]:
```suggestion
```

File: apps/java-api/src/main/java/org/scoalaonline/api/service/SupervisorService.java [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+      }
+      if (supervisor.getEmail() != null) {
+        supervisorToUpdate.setEmail(supervisor.getEmail());
+      }
+      if (supervisor.getDepartment() != null) {
+        supervisorToUpdate.setDepartment(supervisor.getDepartment());
+      }
+      return supervisorRepo.save(supervisorToUpdate);
+    } catch (NoSuchElementException e) {
+      System.out.println("The given id doesn't link to any supervisor.");
```
Comment (CristiPV) [inline apps/java-api/src/main/java/org/scoalaonline/api/service/SupervisorService.java]:
```suggestion
      System.out.println("The supervisor to update doesn't exist.");
```

**Review by CristiPV — CHANGES_REQUESTED** [review]

**Comment by Vimpat** [comment]
¯\_(ツ)_/¯

**Comment by sebastianbadea197** [comment]
¯\\_(ツ)_/¯

**Review by CristiPV — APPROVED** [review]

**Pull request merged**